src/main/java/com/webank/weid/service/impl/EvidenceServiceImpl.java

```diff
@@ -249,6 +254,62 @@ private String getSignerFromPrivKey(List<String> signers, String privateKey) {
         }
     }
 
+    /* (non-Javadoc)
+     * @see com.webank.weid.rpc.generateHash
+     * #generateHash(T object)
+     */
+    @Override
+    public <T> ResponseData<HashString> generateHash(T object) {
+        if (object == null) {
+            return new ResponseData<>(null, ErrorCode.ILLEGAL_INPUT);
+        }
+        if (object instanceof Hashable) {
+            ResponseData<String> hashResp = getHashValue((Hashable) object);
+            if (StringUtils.isEmpty(hashResp.getResult())) {
+                return new ResponseData<>(null, hashResp.getErrorCode(),
+                    hashResp.getErrorMessage());
+            }
+            return new ResponseData<>(new HashString(hashResp.getResult()), ErrorCode.SUCCESS);
+        }
+        if (object instanceof File) {
+            // This will convert all types of file into String stream
+            String rawData = convertFileToString((File) object);
+            if (StringUtils.isEmpty(rawData)) {
+                logger.error("Failed to convert file into String: {}", ((File) object).getName());
+                return new ResponseData<>(null, ErrorCode.ILLEGAL_INPUT);
+            }
+            return new ResponseData<>(new HashString(DataToolUtils.sha3(rawData)),
+                ErrorCode.SUCCESS);
+        }
+        if (object instanceof String) {
+            if (StringUtils.isEmpty((String) object)) {
+                logger.error("Input String is blank, ignored..");
+                return new ResponseData<>(null, ErrorCode.ILLEGAL_INPUT);
+            }
+            return new ResponseData<>(new HashString(DataToolUtils.sha3((String) object)),
+                ErrorCode.SUCCESS);
+        }
+        // last-resort -- use built-in method and default serializer
+        String serialized;
+        try {
+            serialized = DataToolUtils.serialize(object);
+        } catch (Exception e) {
+            logger.info("Failed to serialize object - use built-in hashCode for {}",
+                object.getClass().getCanonicalName());
+            serialized = String.valueOf(object.hashCode());
```

Comment:
Please don't use a default implementation to generate a hash for the types that you don't handle in your function, which may makes some undefined problems as we discussed in wechat work.